R/boxr__internal_misc.R

```diff
@@ -376,3 +376,57 @@ box_terminal_http_codes <- function() {
   )
 }
 
+#' prepare a list 
+#' 
+#' takes a list:
+#'   - changes `NULL` to `NA_character`
+#'   - wraps `list` in another `list`
+#' 
+#' this is motivated by wanting to stack these into a tibble,
```

Comment:
Wondering if we really want 3 smaller functions for this single list stacking behavior, thinking about overhead for trace-backs, we only use `stack_rows()`.

For sequential steps I'd prefer the pipe operator for readability, what are your thoughts?

Reply:
The three functions came about because I wanted to test each of them separately - I wanted to be very sure about the "innermost" functions. 

I can see merging them into a single function at some point, once we are very sure they are doing exactly what we want them to be doing. I'd like to use these functions for more cases, perhaps as the basis for the proposed `box_wrangle()` function which we could use for the existing classes. If so, I suspect having separate functions will help the debugging.

I agree on the pipe for readability - but if it errors, we sacrifice readability in the tracebacks (for now*). 

On overhead, I think the API calls will always be orders of magnitude slower that whatever we are doing to post-process.

*did you see there are changes afoot for the pipe?

Reply:
Yes! the v2.0 changes was what gave me the confidence to suggest it. I know people have poo-poo'd using it in package code in the past, but it seems like the new release will address a folks' problems with it.  While the tracebacks can be a little trickey if chains get long I've also experience the converse where I have to chase down the source code for one more internal function (then another, then another) trying to debug when a package function breaks.